### PR TITLE
feat: display dataflow endpoint authentication details and status

### PR DESCRIPTION
**Pull request by c-ryan-k**
Adds the following displays and evals to dataflow CR checks:
- Displays for various endpoint authentication types
- Displays and evals for Provisioning Status and Runtime Status for all Dataflow CRs

Also tweaks the title of summary checks from "AIO components" to "Service summary checks"

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: azext_edge/edge/providers/check/dataflow.py [hunk 1]
```diff
@@ -342,16 +483,87 @@ def _process_dataflow_destinationsettings(
...
+
+    # display auth method
     check_manager.add_display(
         target_name=target,
         namespace=namespace,
         display=basic_property_display(label="Authentication Method", value=auth_method, padding=padding),
     )
-    # TODO - add displays for various auth types (refs, identity names, etc)
+
+    # show details for various auth methods
```
Comment (Elsie4ever) [inline azext_edge/edge/providers/check/dataflow.py:553]: want to double check, I see a secretRef property, if that's the Kubernetes secret, should we add a check for that?

File: azext_edge/edge/providers/check/dataflow.py [hunk 2]
```diff
@@ -342,16 +483,87 @@ def _process_dataflow_destinationsettings(
...
+        DataFlowEndpointAuthenticationType.sasl.value: {
+            "key": "saslSettings",
+            "displays": [
+                ("Type", "saslType"),
+                ("Secret Ref", "secretRef"),
+            ],
+        },
+        DataFlowEndpointAuthenticationType.anonymous.value: {
+            "key": "anonymousSettings",
+            "displays": [],
```
Comment (Elsie4ever) [inline azext_edge/edge/providers/check/dataflow.py:530]: is there no property under anonymousSettings?

File: azext_edge/edge/providers/check/dataflow.py [hunk 3]
```diff
@@ -54,6 +55,146 @@
...
+        provisioning_status_log_errors = provisioning_status.get("logErrors")
+        if detail_level > ResourceOutputDetailLevel.summary.value and provisioning_status_log_errors:
+            check_manager.add_display(
+                target_name=target_name,
+                namespace=namespace,
+                display=Padding("Log Errors:", (0, 0, 0, inner_padding)),
+            )
+
+        # operation ID on detail level 2
+        provisioning_status_operation_id = provisioning_status.get("operationId")
```
Comment (Elsie4ever) [inline azext_edge/edge/providers/check/dataflow.py:141]: we were talking about this offline, if we want to have this operation id, wondering if making the value into its own line would be better visually?

ex:
Opertation ID:
    abcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcd...

File: azext_edge/edge/providers/check/dataflow.py (hunk 2 above)
Comment (c-ryan-k) [reply azext_edge/edge/providers/check/dataflow.py:530]: Currently there are no defined properties - it's just an empty object with no description in the CRD
image

File: azext_edge/edge/providers/check/dataflow.py (hunk 3 above)
Comment (c-ryan-k) [reply azext_edge/edge/providers/check/dataflow.py:141]: It's very awkward to display - chatting offline to determine whether this is even worth displaying or just removing from the output.

File: azext_edge/edge/providers/check/dataflow.py (hunk 1 above)
Comment (c-ryan-k) [reply azext_edge/edge/providers/check/dataflow.py:553]: We should eventually - there are a few other refs in this set of checks (schemaref, secretref, persistentvolumeclaimref, etc) that likely need validation - I can address those in a future PR that adds a way for us to verify various "refs" exist

Comment (Elsie4ever) [reply azext_edge/edge/providers/check/dataflow.py:559]: sounds good

**Review by Elsie4ever — APPROVED** [review]

**Pull request merged**